### DISCUSSION TITLE
Use in jest tests

### DISCUSSION THREAD

**Discussion by marksmall** [abereghici/remix-themes#9] [General]
For those that are having a similar problem to me, I've been trying to write a **Jest unit test** for my `theme-switcher` component, that uses the `useTheme` hook. I initially was getting the error:

`TypeError: window.matchMedia is not a function`

If you google on this you can see there is some [documentation ](https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom). I added this to my test and got further, but again, got another error, this time `TypeError: Cannot read properties of undefined (reading 'addEventListener')`. I therefore amended the solution from the docs to be:

```JavaScript
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: (query: string) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: () => ({}), // deprecated
    removeListener: () => ({}), // deprecated
    addEventListener: () => ({}),
    removeEventListener: () => ({}),
    dispatchEvent: () => ({}),
  }),
});
```
Using this in the test file, or in some `setupTests.ts|js` file, should allow you to work with the `ThemeProvider` in your tests. e.g.

```JavaScript
render(
      <ThemeProvider specifiedTheme="light" themeAction="/action/set-theme">
        <ThemeSwitcher />,
      </ThemeProvider>,
    );
```

Hope this helps anyone suffering the same trouble as I was.

**Comment by abereghici** [comment]
@marksmall Thank you!

**Comment by abereghici** [comment]
I added unit tests to the library. If you're interested in how to  mock `window.matchMedia` using `vitest`, here is an example: https://github.com/abereghici/remix-themes/blob/main/packages/remix-themes/src/theme-provider.test.tsx#L79
